tests/unit/test_bo.py

```diff
@@ -122,3 +123,113 @@ def test_acquisition():
     assert new.shape == (n2, n_params)
     assert np.all((new[:, 0] >= bounds['a'][0]) & (new[:, 0] <= bounds['a'][1]))
     assert np.all((new[:, 1] >= bounds['b'][0]) & (new[:, 1] <= bounds['b'][1]))
+
+
+class Test_MaxVar:
+    """Run a collection of tests for the MaxVar acquisition."""
+
+    def test_acq_bounds(self, acq_maxvar):
+        """Check if the acquisition is performed within the bounds.
+
+        Parameters
+        ----------
+        acq_maxvar : MaxVar
+            Acquisition method.
+
+        Notes
+        -----
+        - acq_maxvar acquires points from a 2-D Gaussian noise model.
+
+        """
+        bounds = acq_maxvar.model.bounds
+        batch_size = 2
+        n_it = 2
+
+        # Acquiring points.
+        for it in range(n_it):
+            batch_theta = acq_maxvar.acquire(n=batch_size, t=it)
+
+        # Checking if the acquired points are within the bounds.
+        assert np.all(batch_theta >= bounds[0][0])
+        assert np.all(batch_theta <= bounds[0][1])
+
+    def test_gradient(self, acq_maxvar):
+        """Compare the gradients obtained by finite difference and evaluate_gradient.
+
+        Parameters
+        ----------
+        acq_maxvar : MaxVar
+            Acquisition Method.
+
+        """
+        # MANUAL SWITCH: Enable or disable the visualisation of the gradients.
+        vis = False
+
+        # Acquiring some points to initialise the acquisition method's params.
+        n_pts_acq = 10
+        acq_maxvar.acquire(n_pts_acq)
+
+        # Partitioning the axes.
+        bounds = acq_maxvar.model.bounds
+        n_pts = 50
+        lins_dim1, d_dim1 = np.linspace(*bounds[0], num=n_pts, retstep=True)
+        lins_dim2, d_dim2 = np.linspace(*bounds[1], num=n_pts, retstep=True)
+
+        # Computing the gradient using the acquisition method's class.
+        evals = np.zeros(shape=(n_pts, n_pts))
+        grads_maxvar = np.zeros(shape=(2, n_pts, n_pts))
+        for idx_dim1, coord_dim1 in enumerate(lins_dim1):
+            for idx_dim2, coord_dim2 in enumerate(lins_dim2):
+                coord = coord_dim1, coord_dim2
+                evals[idx_dim1, idx_dim2] = acq_maxvar.evaluate(coord)[0]
+                grads_maxvar[:, idx_dim1, idx_dim2] \
+                    = acq_maxvar.evaluate_gradient(coord)[0]
+
+        # Computing the gradient via a finite difference method.
+        grads_np = np.gradient(evals, d_dim1, d_dim2)
+        grads_np = np.array(grads_np)
+
+        if vis:
```

Comment:
Remove this block. These tests are supposed to be fully automatic (even if you'd have to separately switch it on). Plotting can be done in local scripts.

Reply:
I have written an alternative test for the gradient function. It is much more concise thus easier to understand:
`
    def test_gradient_abstraction(self, acq_maxvar):
        """Test the gradient function using GPy's GradientChecker.

        Parameters
        ----------
        acq_maxvar : MaxVar
            Acquisition method.

        """
        from GPy.models.gradient_checker import GradientChecker
        n_pts_test = 100
        n_dim_fixture = len(acq_maxvar.model.bounds)

        checker_grad = GradientChecker(acq_maxvar.evaluate,
                                       acq_maxvar.evaluate_gradient,
                                       np.random.randn(n_pts_test, n_dim_fixture))
        # The tolerance corresponds to the allowed deviation from the unity of
        # the ratio between analytical and numerical gradients.
        assert checker_grad.checkgrad(tolerance=1e-4)
`
Essentially it does the same things with an implicit testing abstraction provided by GPy. Thoughts on having both of the tests or replacing the long one with the shorter one?

Reply:
No reason to have both. Which ever is faster?